sql/core/src/main/scala/org/apache/spark/sql/execution/joins/BroadcastHashJoinExec.scala

```diff
@@ -60,6 +62,66 @@ case class BroadcastHashJoinExec(
     }
   }
 
+  override def outputPartitioning: Partitioning = {
+    joinType match {
+      case _: InnerLike =>
+        streamedPlan.outputPartitioning match {
+          case h: HashPartitioning => PartitioningCollection(expandOutputPartitioning(h))
+          case c: PartitioningCollection =>
+            def expand(partitioning: PartitioningCollection): Partitioning = {
```

Comment:
Could you pull out this inner metdhod and define it outside as private? Also, we need to assign a resonable method name.

Reply:
I overloaded `expandOutputPartitioning` so that it can take either `HashPartitioning` or `PartitioningCollection` and returns `PartitioningCollection`.